platforms/core-runtime/base-services/src/main/java/org/gradle/internal/service/ServiceScopeValidatorWorkarounds.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.service;
+
+import org.gradle.api.NonNullApi;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+@NonNullApi
+public class ServiceScopeValidatorWorkarounds {
+
+    private static final Set<String> SUPPRESSED_VALIDATION_CLASSES = new HashSet<String>(Arrays.asList(
```

Comment:
💅 An array of strings (or even just a list) would be a little simpler here

Reply:
I'd prefer to keep this as a `Set`. Using an immutable version seems reasonable and even more concise 👍